src/elements/content-uploader/ContentUploader.js

```diff
@@ -304,26 +304,27 @@ class ContentUploader extends Component<Props, State> {
             newItemIds[getFileId(file, rootFolderId)] = true;
         });
 
-        this.setState(state => ({
-            itemIds: {
-                ...state.itemIds,
-                ...newItemIds,
-            },
-        }));
-
         clearTimeout(this.resetItemsTimeout);
 
         const firstFile = getFile(newFiles[0]);
 
-        // webkitRelativePath should be ignored when the upload destination folder is known
-        if (firstFile.webkitRelativePath && !isRelativePathIgnored) {
-            this.addFilesWithRelativePathToQueue(newFiles, itemUpdateCallback);
-            return;
-        }
-
-        onBeforeUpload(newFiles);
-
-        this.addFilesWithoutRelativePathToQueue(newFiles, itemUpdateCallback);
+        this.setState(
+            state => ({
+                itemIds: {
+                    ...state.itemIds,
+                    ...newItemIds,
+                },
+            }),
+            () => {
+                if (firstFile.webkitRelativePath && !isRelativePathIgnored) {
+                    // webkitRelativePath should be ignored when the upload destination folder is known
+                    this.addFilesWithRelativePathToQueue(newFiles, itemUpdateCallback);
+                } else {
+                    onBeforeUpload(newFiles);
```

Comment:
Doesn't this function needed to be executed in either case, since it is a hook or renderProp that is passed into this component?

Reply:
I agree your idea but the logic of the original code is like my code now. I think it might be a mistake in the original code.